linalg/sparsemat.hpp

```diff
@@ -578,6 +578,16 @@ class SparseMatrix : public AbstractSparseMatrix
    Type GetType() const { return MFEM_SPARSEMAT; }
 };
 
+inline std::ostream& operator<<(std::ostream& os, SparseMatrix const& mat)
+{
+   // HostRead forces synchronization
+   mat.HostReadI();
+   mat.HostReadJ();
+   mat.HostReadData();
+   mat.Print(os);
```

Comment:
Should `SparseMatrix::Print` force synchronization?

Reply:
Yes, otherwise you get random values if assembled on GPU, that happened to me...

Reply:
I see you want to add this synchronization, inside `Print()`, I would agree that it would be safer than doing it only here. @tzanio @v-dobrev What do you think?

Reply:
done.